src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1119,17 +1120,21 @@ public Solution WithProjectAnalyzerReferences(ProjectId projectId, IEnumerable<A
 
         private Solution AddDocument(DocumentState state)
         {
-            if (state == null)
+            var notificationService = this.Workspace.Services.GetService<IGlobalOperationNotificationService>();
+            using (notificationService?.Start("Remove Document"))
             {
-                throw new ArgumentNullException(nameof(state));
-            }
+                if (state == null)
+                {
+                    throw new ArgumentNullException(nameof(state));
+                }
 
-            CheckContainsProject(state.Id.ProjectId);
+                CheckContainsProject(state.Id.ProjectId);
```

Comment:
hmmm. there should be some project load event from vs. let me take a quick look.